core/test/Test/Pos/Core/Bi.hs

```diff
@@ -0,0 +1,1739 @@
+
+golden_UpId :: Property
+golden_UpId = goldenTestBi exampleUpId "test/golden/UpId"
+
+roundTripUpId :: Property
+roundTripUpId = eachOf 20 (feedPM genUpId) roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- UpsData NB: UpsData is not a type it is a record accessor of `UpdateProposalToSign`
+--------------------------------------------------------------------------------
+
+roundTripUpsData :: Property
+roundTripUpsData = eachOf 20 genUpsData roundTripsBiShow
+
+--------------------------------------------------------------------------------
+-- VoteId
+--------------------------------------------------------------------------------
+
+golden_VoteId :: Property
+golden_VoteId = goldenTestBi vID "test/golden/VoteId"
+    where
+        vID = (exampleUpId, examplePublicKey, False)
+-- ```type VoteId = (UpId, PublicKey, Bool)```
+
+roundTripVoteId :: Property
+roundTripVoteId = eachOf 20 (feedPM genVoteId) roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- VssCertificate
+--------------------------------------------------------------------------------
+
+golden_VssCertificate :: Property
+golden_VssCertificate = goldenTestBi exampleVssCertificate "test/golden/VssCertificate"
+
+roundTripVssCertificate :: Property
+roundTripVssCertificate = eachOf 10 (feedPM genVssCertificate) roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- VssCertificatesHash
+--------------------------------------------------------------------------------
+
+golden_VssCertificatesHash :: Property
+golden_VssCertificatesHash = goldenTestBi (exampleVssCertificatesHash 10 4) "test/golden/VssCertificatesHash"
+
+roundTripVssCertificatesHash :: Property
+roundTripVssCertificatesHash = eachOf 10 (feedPM genVssCertificatesHash) roundTripsBiBuildable
+
+--------------------------------------------------------------------------------
+-- VssCertificatesMap
+--------------------------------------------------------------------------------
+
+golden_VssCertificatesMap :: Property
+golden_VssCertificatesMap = goldenTestBi (exampleVssCertificatesMap 10 4) "test/golden/VssCertificatesMap"
+
+roundTripVssCertificatesMap :: Property
+roundTripVssCertificatesMap = eachOf 10 (feedPM genVssCertificatesMap) roundTripsBiShow
+
+--------------------------------------------------------------------------------
+-- Helpers
+--------------------------------------------------------------------------------
+
+feedPM :: (ProtocolMagic -> H.Gen a) -> H.Gen a
+feedPM genA = genA =<< genProtocolMagic
+
+feedPC :: (ProtocolConstants -> H.Gen a) -> H.Gen a
+feedPC genA = genA =<< genProtocolConstants
+
+feedPMC :: (ProtocolMagic -> ProtocolConstants -> H.Gen a) -> H.Gen a
+feedPMC genA = do pm <- genProtocolMagic
+                  pc <- genProtocolConstants
+                  genA pm pc
```

Comment:
The formatting elsewhere in this code base has the `do` on the same line as the function name and parameter and the rest on the following lines, indented by 4 spaces.

Be nice to keep the code reasonably consistent.